packages/component/src/providers/RovingTabIndex/RovingTabIndexComposer.tsx

```diff
@@ -109,10 +113,10 @@ const RovingTabIndexComposer: FC<RovingTabIndexContextProps> = ({ children, onEs
             const nextIndex = itemIndices.indexOf(value) - 1;
 
             if (nextIndex < 0) {
-              return itemIndices[itemIndices.length - 1];
+              return itemIndices[itemIndices.length - 1] as number;
             }
 
-            return itemIndices[+nextIndex];
+            return itemIndices[+nextIndex] as number;
```

Comment:
Instead of casting `as number` you could consider using a type guard. 

```js
function isNumber(value: unknown): value is number {
  return typeof value === 'number';
}
```
```js
if (!isNumber(value)) { throw new Error(...); }
```